shared_model/interfaces/iroha_internal/block_variant.hpp

```diff
@@ -12,10 +12,66 @@
 namespace shared_model {
   namespace interface {
 
-    /// Block variant with either block or empty block
-    using BlockVariantType =
-        boost::variant<std::shared_ptr<shared_model::interface::Block>,
-                       std::shared_ptr<shared_model::interface::EmptyBlock>>;
+    class BlockVariant
+        : public boost::variant<
+              std::shared_ptr<shared_model::interface::Block>,
+              std::shared_ptr<shared_model::interface::EmptyBlock>>,
+          protected AbstractBlock {
+     private:
+      using VariantType =
+          boost::variant<std::shared_ptr<shared_model::interface::Block>,
+                         std::shared_ptr<shared_model::interface::EmptyBlock>>;
+
+     public:
+      using VariantType::VariantType;
+
+      interface::types::HeightType height() const override {
+        return iroha::visit_in_place(
+            *this, [](const auto &any_block) { return any_block->height(); });
+      }
+
+      const interface::types::HashType &prevHash() const override {
+        return std::move(iroha::visit_in_place(
```

Comment:
Why move here?

Reply:
It seems that there is no case when `return std::move` is applicable https://stackoverflow.com/a/14856553

Reply:
That is because lambda returns value instead of reference, because it's return type is deduced with `auto` instead of `decltype(auto)` by default. You can manually specify lambda return as `-> decltype(auto)`.